controlplane/src/core/controllers/scim.ts

```diff
@@ -0,0 +1,408 @@
+import { PostgresJsDatabase } from 'drizzle-orm/postgres-js';
+import { FastifyPluginCallback, FastifyRequest } from 'fastify';
+import fp from 'fastify-plugin';
+import * as schema from '../../db/schema.js';
+import { OrganizationRepository } from '../repositories/OrganizationRepository.js';
+import { UserRepository } from '../repositories/UserRepository.js';
+import ApiKeyAuthenticator, { ApiKeyAuthContext } from '../services/ApiKeyAuthenticator.js';
+import Keycloak from '../services/Keycloak.js';
+
+export type ScimControllerOptions = {
+  db: PostgresJsDatabase<typeof schema>;
+  organizationRepository: OrganizationRepository;
+  userRepository: UserRepository;
+  authenticator: ApiKeyAuthenticator;
+  keycloakClient: Keycloak;
+  keycloakRealm: string;
+};
+
+declare module 'fastify' {
+  interface FastifyRequest {
+    authContext: ApiKeyAuthContext;
+  }
+}
+
+const plugin: FastifyPluginCallback<ScimControllerOptions> = function Scim(fastify, opts, done) {
```

Comment:
Please add integration tests for this plugin.

Reply:
As far I remember, there are concrete acceptance tests for SCIM integrations. It would be great, if we could reference them in the integration tests.